code/modules/mob/living/carbon/carbon.dm

```diff
@@ -671,7 +671,7 @@
 	if(status_flags & GODMODE)
 		return
 	if(stat != DEAD)
-		if(health<= HEALTH_THRESHOLD_DEAD || !getorgan(/obj/item/organ/brain))
+		if(health<= HEALTH_THRESHOLD_DEAD)

```

Comment:
Yeah. Organ's Remove() specifically checks whether an organ's vital var is true, which the brain is, and if so called the mob's death(). Remove is called in both decapitation and surgery.